### DISCUSSION TITLE
Current track not shown when playing album

### DISCUSSION THREAD

**Discussion by Malvineous** [librespot-org/librespot#947] [General]
I'm not sure whether this is a bug in librespot or in the Spotify web client, so just curious whether anyone else has seen it.

If I play songs in a playlist through the web player, then the current song is highlighted green, and as I skip between tracks the green highlight moves around to show the currently playing track, as expected.

This works when using only the web player, on both playlists and when playing albums directly without adding them to a playlist first.

However when I switch to librespot as the active playback device, the green highlight disappears, but only when playing albums directly.  If I am playing a playlist, the green highlight works fine.  If I play any track on any album that isn't in a playlist, the green highlight disappears.  It returns if I switch the active audio device to something other than librespot, but switching back to librespot causes the green highlight to disappear again.

I have tried using the new API and the problem remains.  While playlists work fine, I can only get the green highlight to function normally when playing an album if librespot is NOT the active playback device.  When librespot becomes the active playback device, the green highlight disappears, but only on albums.  It works fine on playlists.

Is anyone else seeing this behaviour?

I have been seeing it for quite some time (months) but I only thought to mention it now in case it's something the new API might be able to fix.

**Comment by roderickvd** [comment]
Yes this is a long-standing issue (though I don't think anyone ever filed an issue for it). At this point it is not clear to me what causes this. Because I need to investigate anyway, I'd rather do that when I have implemented the dealer on `new-api` and spend time investigating that instead of the old API. This is mentioned on the todo list: #917.
